### PR TITLE
Convert infrastructure to utilize bootstrapper

### PR DESCRIPTION
**Pull request by kpdecker**
This effectively creates one source of truth on file and module resolution,
preventing potential issues where module a depends on {b,c} and b depends on
{c} and both hardcode different versions, leading to indeterminate behavior at
runtime.

Additionally moves all of the loading logic into the initial entry point so
there is minimal duplication across different components in the system.

Users are now expected to either load the bootstrap.js file in their site or
to utilize the `output.bootstrap` flag which will inline the bootstrap logic
into the main entry point. For cases where this is inlined, no other apps can
link to this code as there are sanity checks in place to prevent multiple
circus bootstraps being loading on the same page as they will conflict on a
few global variables.

This has the added bonus of allowing us to rewrite dependencies of files at
publish time, allowing us to defer the local vs. not decision across the
entire app. (Those changes to be finalized in a future commit)

### CODE REVIEW

File: lib/client/bootstrap.js.hbs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    jsPaths = {{jsChunkNames}},
+    cssPaths = {{cssChunkNames}};
+
+var isArray = Array.isArray || function(arg) {
+  return Object.prototype.toString.call(arg) === '[object Array]';
+};
+
+function resolveUrl(url) {
+    // Relative URLs should honor the path prefix
```
Comment (ryan-roemer) [inline lib/client/bootstrap.js.hbs]: NIT: extra space in front of comment.

**Comment by kpdecker** [comment]
/cc @Maciek416 when this PR is released, you will also have the arbitrary lookup functionality that you need. Something like:

```
function loadIt(name) {
  for (var componentName in __webpack_components__.ex) {
    if (__webpack_components__.ex[name]) {
      return __webpack_components__.c(name);
    }
  } 
}
```

Should retrieve any exported modules that are loaded and exist in the page (You need to do some more work there if you want to ensure that they are loaded first, but that too is doable)

File: lib/client/bootstrap.js.hbs (hunk 1 above)
Comment (kpdecker) [inline lib/client/bootstrap.js.hbs]: Removed.

**Comment by kpdecker** [comment]
Fixes #17

**Comment by kpdecker** [comment]
Released in 2.0.0

**Pull request merged**